src/drive/web/modules/drive/Toolbar/components/ShortcutCreationModal.jsx

```diff
@@ -54,8 +54,25 @@ const ShortcutCreationModal = ({ onClose, onCreated, displayedFolder }) => {
         Alerter.success(t('Shortcut.created'))
         if (onCreated) onCreated()
         onClose()
-      } catch (e) {
-        Alerter.error(t('Shortcut.errored'))
+      } catch (error) {
+        if (
+          error.message.includes(
+            'Invalid filename containing illegal character(s):'
```

Comment:
As we maintain CozyClient, I think it would be great to type our errors. For example here we could have
```js
class CozyInvalidCharacterError extends Error {
  constructor(errorMessage, invalidChars) {
    // store custom properties
  }
}
```
And it would be consumed like this
```js
try {
} catch (e) {
  if (e instance of CozyInvalidCharacterError) {
    // our specific code
  }
}
```

Also this would reduce the number of `Invalid filename containing illegal character(s)` string declaration on cozy-client side.

But this seems not to be used a lot at Cozy. @Crash-- do you have any opinion on this?

Reply:
This is not the way we do it ATM. I don't know if I'm more convinced by creating custom classes for each of our errors or if we add something like "app error code". 

So let's move forward with the current implementation based on the string manipulation, and let's discuss about errors in an CozyClient's issue. 👍

Reply:
Here it is : https://github.com/cozy/cozy-drive/issues/2445

Reply:
I think we can move this issue to CozyClient, it's more relevant there ;). 

Also, take a look at FetchError from CozyClient

Reply:
Holy potato, It was supposed to be on client, how am I able to do this kind of mistake with so few opened tabs?? 😅 

Issue transfered https://github.com/cozy/cozy-client/issues/1088